lcls_tools/common/data_analysis/fit/methods.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from scipy.stats import norm, gamma
+from lcls_tools.common.data_analysis.fit.method_base import MethodBase
+
+
+class GaussianModel(MethodBase):
+    """
+    GaussianModel Class that finds initial parameter values for gaussian distribution
+    and builds probability density functions for the likelyhood a parameter
+    to be that value based on those initial parameter values.
+    Passing this class the variable profile_data automatically updates
+    the initial values and and probability density functions to match that data.
+    """
+
+    param_names: list = ["mean", "sigma", "amplitude", "offset"]
+    param_bounds: np.ndarray = np.array(
+        [
+            [0.01, 1.0],
+            [0.01, 5.0],
+            [0.01, 1.0],
+            [0.01, 1.0],
+        ]
+    )
+
+    def find_init_values(self) -> dict:
+        """Fit data without optimization, return values."""
+
+        data = self._profile_data
+        x = np.linspace(0, 1, len(data))
+        # init_fit = norm.pdf(data)
+        offset = data.min()
+        amplitude = data.max() - offset
+
+        weighted_mean = np.average(x, weights=data)
+        weighted_sigma = np.sqrt(np.cov(x, aweights=data))
+
+        self.init_values = {
+            self.param_names[0]: weighted_mean,  # data.mean()
+            self.param_names[1]: weighted_sigma,  # data.std()
+            self.param_names[2]: amplitude,
+            self.param_names[3]: offset,
+        }
+        return self.init_values
+
+    def find_priors(self) -> dict:
```

Comment:
The Bayesian approach makes sense for ML, but it might confuse the rest of the LCLS Tools users. We should refactor to make these parameters more readily accessible.

Reply:
Do you mean pulling out the "priors" and putting that into another file?

Reply:
I think when Nicole asked for fitting she was expecting a function that takes in position and intensity data and outputs relevant beam parameters, like position, beam width, etc. I think we could solve this discrepancy between ML and physics users by creating a method using this model to compute beam parameters with the appropriate parameter labels and units.

We can talk about this more during the meeting today.